src/transformers/tokenization_utils.py

```diff
@@ -1199,7 +1199,13 @@ def get_input_ids(text):
 
         input_ids = []
         for ids_or_pair_ids in batch_text_or_text_pairs:
-            if isinstance(ids_or_pair_ids, (list, tuple)) and len(ids_or_pair_ids) == 2:
+            is_list_of_untokenized_strings = (
+                isinstance(ids_or_pair_ids, (list, tuple))
+                and len(ids_or_pair_ids) > 0
```

Comment:
Couldn't this `len(ids_or_pair_ids) > 0` be set to `len(ids_or_pair_ids) == 2`, since this edge case only happens when there are exactly two strings?

This would prevent the interpreter from evaluating the third statement which currently needs to tokenize each token in `ids_or_pair_ids` when `len(ids_or_pair_ids) == 1 and len(ids_or_pair_ids) > 2`

Reply:
Yeah, I guess. I wanted to keep the boolean `is_list_of_untokenized_strings` as general as possible in case there will be more changes, but I think it's better to go for better speed here since it's a very general function!

Reply:
K no need for any tricks anymore - the `is_pretokenized` flag is much easier.